install.sh

```diff
@@ -1137,9 +1302,15 @@ sandcats_register_name() {
     return
   fi
 
-  # Validate the client-side, to avoid problems, against the same
-  # regex that the server is using.
-  if ! [[ $DESIRED_SANDCATS_NAME =~ ^[0-9a-zA-Z]{1,20}$ ]] ; then
+  # If the user wants help, offer help.
+  if [ "help" = "$DESIRED_SANDCATS_NAME" ] ; then
+    sandcats_provide_help
+    return
+  fi
+
+  # Validate the client-side, to avoid problems, against a slightly
+  # less rigorous regex than the server is using.
+  if ! [[ $DESIRED_SANDCATS_NAME =~ ^[0-9a-zA-Z-]{1,20}$ ]] ; then
```

Comment:
Just a note for reviewers: the change here is that I updated this code to allow hyphens anywhere; the server checks our hyphen use against a more rigorous regexp.